subprojects/core/src/main/java/org/gradle/api/tasks/Exec.java

```diff
@@ -82,68 +85,79 @@ public Exec setArgs(@Nullable Iterable<?> arguments) {
      * {@inheritDoc}
      */
     @Override
-    public Exec setIgnoreExitValue(boolean ignoreExitValue) {
-        return super.setIgnoreExitValue(ignoreExitValue);
+    @ReplacesEagerProperty(adapter = Exec.StandardInputAdapter.class)
+    public Property<InputStream> getStandardInput() {
+        return super.getStandardInput();
     }
 
     /**
      * {@inheritDoc}
      */
+    @Internal
     @Override
-    @ToBeReplacedByLazyProperty
-    public InputStream getStandardInput() {
-        return super.getStandardInput();
+    @ReplacesEagerProperty(adapter = Exec.StandardOutputAdapter.class)
+    public Property<OutputStream> getStandardOutput() {
+        return super.getStandardOutput();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public Exec setStandardInput(InputStream inputStream) {
-        return super.setStandardInput(inputStream);
+    @ReplacesEagerProperty(adapter = Exec.ErrorOutputAdapter.class)
+    public Property<OutputStream> getErrorOutput() {
+        return super.getErrorOutput();
     }
 
     /**
-     * {@inheritDoc}
+     * No need to upgrade getter since it's already upgraded via BaseExecSpec
      */
-    @Internal
-    @Override
-    @ToBeReplacedByLazyProperty
-    public OutputStream getStandardOutput() {
-        return super.getStandardOutput();
+    static class IgnoreExitValueAdapter {
+        @BytecodeUpgrade
+        static Exec setIgnoreExitValue(Exec task, boolean value) {
+            task.getIgnoreExitValue().set(value);
+            return task;
+        }
     }
 
     /**
-     * {@inheritDoc}
+     * No need to upgrade getter since it's already upgraded via BaseExecSpec
      */
-    @Override
-    public Exec setStandardOutput(OutputStream outputStream) {
-        return super.setStandardOutput(outputStream);
+    static class StandardInputAdapter {
+        @BytecodeUpgrade
+        static Exec setStandardInput(Exec task, InputStream value) {
+            task.getStandardInput().set(value);
+            return task;
+        }
     }
 
     /**
-     * {@inheritDoc}
+     * No need to upgrade getter since it's already upgraded via BaseExecSpec
      */
-    @Override
-    @ToBeReplacedByLazyProperty
-    public OutputStream getErrorOutput() {
-        return super.getErrorOutput();
+    static class StandardOutputAdapter {
+        @BytecodeUpgrade
+        static Exec setStandardOutput(Exec task, OutputStream value) {
+            task.getStandardOutput().set(value);
+            return task;
+        }
     }
 
     /**
-     * {@inheritDoc}
+     * No need to upgrade getter since it's already upgraded via BaseExecSpec
      */
-    @Override
-    public Exec setErrorOutput(OutputStream outputStream) {
-        return super.setErrorOutput(outputStream);
+    static class ErrorOutputAdapter {
+        @BytecodeUpgrade
+        static Exec setErrorOutput(Exec task, OutputStream value) {
+            task.getErrorOutput().set(value);
+            return task;
+        }
```

Comment:
🤔 I just realized that `self` could have been a good name to use for these `this`-parameters. Not sure if it's worth renaming the existing ones.

Reply:
I renamed occurences in this class. We can try to be consistent in the future